### PR TITLE
RCC and build rules

### PR DESCRIPTION
**Pull request by felix-salfelder**
see #697

### CODE REVIEW

**Comment by in3otd** [comment]
Travis fails with an unexpected [error message](https://travis-ci.org/Qucs/qucs/jobs/254811205#L3821); restarted it still fails there. But passes on OS X.
Interestingly on this latter `rcc` is called as
`usr/local/bin/rcc -o qucs_.cpp qucs_.qrc`
while on Linux as
`/usr/bin/rcc -o qucs_.cpp ../../../qucs/qucs/qucs_.qrc`

**Comment by felix-salfelder** [comment]
`/usr/bin/rcc -o qucs_.cpp ../../../qucs/qucs/qucs_.qrc` looks perfectly fine to me.

on my machine, it's not complaining...
rcc bug? i have rcc 5.7.1

(don't know whats wrong on OSx though.)

**Comment by in3otd** [comment]
ah, the difference is that on OS X we run `make` (`qucscheck`) and on Linux `make distcheck` instead. And this latter fails also locally here, with the same error message. But no idea why...

BTW, I thought/assumed that `rcc 5.x` was for Qt5.

What about naming the generated file `qucs.qrc.cpp` similarly to what is done with the `.moc` files?

**Comment by felix-salfelder** [comment]
> rcc 5.x was for Qt5.

perhaps it's backwards compatible? the buggy file lookup does not look related to the qt version to me.

> qucs.qrc.cpp

sounds good.

**Comment by in3otd** [comment]
...it's seems we opened another small to Pandora's box: the `make distcheck` are failing because the components bitmaps are not distributed in the tarball (there was no need to, as we distributed the output of `rcc`) so the `.qrc` processing fails since the files to process are not there. 
Then we have other `.qrc` files for the Tools in the sub-directories which would need to be handled in the same way as the main app resource file.

I'm ok to just add the `rcc-qt4` to `configure.ac` for the moment but feel free to handle also the stuff above if you feel so inclined.

**Comment by felix-salfelder** [comment]
> there was no need to

hmm, i think the bitmaps qualify as source, and should be included.

thanks for figuring that out, i was definitely on the wrong track

File: qucs/qucs/Makefile.am [hunk 1]
```diff
@@ -100,15 +104,16 @@ EXTRA_DIST = \
 	bitmaps/qucsrescodes.icns \
 	bitmaps/qucstrans.icns
 
-SUFFIXES = .h .moc.cpp .qrc.cpp
+SUFFIXES = .h .moc.cpp .qrc
 
-.qrc.cpp.cpp:
-qucs.qrc.cpp: qucs.qrc
+.qrc.cpp:
```
Comment (in3otd) [inline qucs/qucs/Makefile.am]: what is this for? Doesn't the *Suffix rules with no recipe are also meaningless* (unlike pattern rules) from the docs apply also here?

Comment (felix-salfelder) [inline qucs/qucs/Makefile.am]: my understanding is, that `.qrc.cpp:` instructs automake to use the path `.qrc -> .cpp -> .o`, where applicable.

cf https://www.gnu.org/software/automake/manual/html_node/Suffixes.html

**Comment by in3otd** [comment]
thanks,
what shall we do with the other `.qrc` files
```
./qucs/qucs-transcalc/qucstrans.qrc
./qucs/qucs-attenuator/qucsattenuator.qrc
./qucs/qucs-activefilter/qucsactivefilter.qrc
./qucs/qucs-filter-v2/qucsfilter.qrc
./qucs/qucs-lib/qucslib.qrc
./qucs/qucs-rescodes/qucsrescodes.qrc
./qucs/qucs-filter/qucsfilter.qrc
```
at a quick glance also there we distribute the compiled file - but also the source `.qrc` file, without the source bitmaps... seems a bit weird to me.

**Comment by in3otd** [comment]
thanks, just a couple of details: in `qucs/stuff.am` you have the GPL blurb set to version 3. Could that file be included also by `qucs/qucs/Makefile.am`, as done for the other makefiles?

**Comment by in3otd** [comment]
one more thing, we would need an updated rule(s) in `.gitignore` for the compiled resource files, we currently have a line `qrc_*.cpp` which of course does not work anymore

**Comment by in3otd** [comment]
looks good, do you plan to squash/rebase/something before merging?

**Comment by in3otd** [comment]
I realized we forgot to update the `CMakefiles` with the new `.qrc` files names, so I've pushed a commit to fix that. It builds fine here with `cmake` now - I'll try later, in another PR, to add a build using `cmake` to Travis.

**Comment by felix-salfelder** [comment]
thanks. did'nt think about that.

i once thought that appveyor runs the cmake system. if that's not the case, why do we need cmake then?

**Comment by in3otd** [comment]
don't remember if we used `cmake` for AppVeyor in the past but currently it also uses the autotools.
I think the `cmake` build was added at some point in the past as a possible alternative and, looking at the old commits comments, because the cmakefiles can be used in Qt Creator but I don't know much more.

**Comment by felix-salfelder** [comment]
in general, id prefer to reduce the clutter.

if someone (here: you) maintains the cmakefiles, i do not object `:)`

**Comment by in3otd** [comment]
yep, don't know how much `cmake` is actually used to build Qucs - adding a `cmake` build to Travis is an attempt to shift the maintenance burden to every developer :grin:

BTW, I'll merge this PR shortly, unless there are any objections...

**Comment by felix-salfelder** [comment]
> shift the maintenance burden to every developer

let's discuss how to get rid of this in another place. it's likely not too urgent for 0.20.

**Pull request merged**